drivers/gpio/gpio_max22190.c

```diff
@@ -0,0 +1,651 @@
+/*
+ * Copyright (c) 2010-2024 Analog Devices Inc.
+ * Copyright (c) 2024 Baylibre SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_max22190_gpio
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(gpio_max22190);
+
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+	#warning "GPIO MAX22190 driver enabled without any devices"
+#endif
+
+#define MAX22190_ENABLE 1
+#define MAX22190_DISABLE 0
+
+#define MAX22190_READ 0
+#define MAX22190_WRITE 1
+
+#define MAX22190_MAX_PKT_SIZE 3
+#define MAX22190_CHANNELS 8
+#define MAX22190_FAULT2_ENABLES 5
+
+#define MAX22190_WB_REG 0x0
+#define MAX22190_DI_REG 0x2
+#define MAX22190_FAULT1_REG 0x4
+#define MAX22190_FILTER_IN_REG(x) (0x6 + (2 * (x)))
+#define MAX22190_CFG_REG 0x18
+#define MAX22190_IN_EN_REG 0x1A
+#define MAX22190_FAULT2_REG 0x1C
+#define MAX22190_FAULT2_EN_REG 0x1E
+#define MAX22190_GPO_REG 0x22
+#define MAX22190_FAULT1_EN_REG 0x24
+#define MAX22190_NOP_REG 0x26
+
+#define MAX22190_CH_STATE_MASK(x) BIT(x)
+#define MAX22190_DELAY_MASK GENMASK(2, 0)
+#define MAX22190_FBP_MASK BIT(3)
+#define MAX22190_WBE_MASK BIT(4)
+#define MAX22190_RW_MASK BIT(7)
+#define MAX22190_ADDR_MASK GENMASK(6, 0)
+#define MAX22190_ALARM_MASK GENMASK(4, 3)
+#define MAX22190_POR_MASK BIT(6)
+
+#define MAX22190_FAULT_MASK(x) BIT(x)
+#define MAX22190_FAULT2_WBE_MASK BIT(4)
+
+#define MAX22190_FAULT2_EN_MASK GENMASK(5, 0)
+
+#define MAX22190_CFG_REFDI_MASK BIT(0)
+#define MAX22190_CFG_CLRF_MASK BIT(3)
+#define MAX22190_CFG_24VF_MASK BIT(4)
+
+#define PRINT_ERR_BIT(bit1, bit2) if ((bit1) & (bit2)) LOG_ERR("[%s] %d", #bit1, bit1)
+
+#define max22190_clean_por(dev) max22190_reg_update(dev, MAX22190_FAULT1_REG, MAX22190_POR_MASK,\
+							 FIELD_PREP(MAX22190_POR_MASK, 0))
+
+enum max22190_ch_state {
+	MAX22190_CH_OFF,
+	MAX22190_CH_ON
+};
+
+enum max22190_ch_wb_state {
+	MAX22190_CH_NO_WB_BREAK,
+	MAX22190_CH_WB_COND_DET
+};
+
+enum max22190_mode {
+	MAX22190_MODE_0,
+	MAX22190_MODE_1,
+	MAX22190_MODE_2,
+	MAX22190_MODE_3,
+};
+
+typedef union _max22190_fault1 {
```

Comment:
Not sure why CI didn't catch this, but new typedefs are discouraged

Reply:
Thanks , noted `typedef` are removed.